src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs

```diff
@@ -324,6 +325,19 @@ protected uint UninstallMsi(string productCode, string logFile, bool ignoreDepen
             throw new InvalidOperationException($"Invalid configuration: elevated: {IsElevated}, client: {IsClient}");
         }
 
+        internal protected string GetWorkloadHistoryDirectory(string sdkFeatureBand)
+        {
+            return Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "dotnet", "workloads", sdkFeatureBand.ToString(), "history");
```

Comment:
We probably need to add the architecture to the path here.  @joeloff, what would be a good path to keep workload history records for MSI-based installs?

Reply:
Good point. I think we should also start doing that for the userlocal file, too.